bson/primitive/objectid.go

```diff
@@ -114,7 +123,11 @@ func (id *ObjectID) UnmarshalJSON(b []byte) error {
 			}
 		}
 
-		if len(str) != 24 {
+		n := len(str)
+		if n == 0 {
```

Comment:
In Extended JSON, an empty objectID is a hex string with all 0s, so the length cannot be 0 and be valid.